packages/imperative/src/security/src/DefaultCredentialManager.ts

```diff
@@ -116,12 +117,13 @@ export class DefaultCredentialManager extends AbstractCredentialManager {
      */
     public async initialize(): Promise<void> {
         try {
-            // Imperative overrides the value of process.mainModule.filename to point to
+            // Imperative overrides the value of require.main.filename to point to
             // our calling CLI. Since our caller must supply keytar, we search for keytar
             // within our caller's path.
             const requireOpts: any = {};
-            if (process.mainModule?.filename != null) {
-                requireOpts.paths = [process.mainModule.filename, ...require.resolve.paths("@zowe/secrets-for-zowe-sdk")];
+            const initPath = ImperativeConfig.instance.callerLocation ?? require.main.filename;
+            if (initPath != null) {
+                requireOpts.paths = [initPath, ...require.resolve.paths("@zowe/secrets-for-zowe-sdk")];
```

Comment:
I was under the impression that was the reason why it falls back to require.main.filename, since if there is no callerLocation, then it just behaves as it did before.

Reply:
Follow up to our earlier discussion - `ImperativeConfig.instance.callerLocation` returns `null` so it would be ok to access in Zowe Explorer. However this may only be because we call `ProfileInfo.initImpUtils` which initializes a fake Imperative config object. Thanks for removing the reference, I think it is the safer option for extenders using our SDKs - will leave this comment unresolved for now to get a second opinion 😋